pkg/rotator/rotator.go

```diff
@@ -521,13 +568,53 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		if err := r.client.Update(r.ctx, vwh); err != nil {
 			return reconcile.Result{Requeue: true}, err
 		}
+
+		// Ensure certs on CRDs.
+		if result, err := r.ensureCRDCerts(artifacts.CertPEM); err != nil {
+			return result, err
+		}
+
+		if !*r.wasCAInjected {
+			*r.wasCAInjected = true
+		}
 	}
 
 	return reconcile.Result{}, nil
 }
 
-// ensureCertsExist ensure the cert files exist.
-func (cr *CertRotator) ensureCertsExist() {
+func (r *ReconcileWH) ensureCRDCerts(certPem []byte) (reconcile.Result, error) {
+	for _, crdName := range r.crdNames {
+		crd := &unstructured.Unstructured{}
+		crd.SetGroupVersionKind(crdGVK)
+		crdKey := types.NamespacedName{Name: crdName}
+		if err := r.client.Get(r.ctx, crdKey, crd); err != nil {
+			if k8sErrors.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{Requeue: true}, err
+		}
+
+		if !crd.GetDeletionTimestamp().IsZero() {
```

Comment:
If the CRD is being deleted, shouldn't we just skip this CRD and move on to the next CRD in the loop?

Reply:
Ah right, will update it.

Reply:
I updated the logic to only return (and only return error) from the loop for transient errors, e.g. cannot get or update, so the request will be re-queued to finish the loop.

For CRD not found, being deleted, or not having caBundle field, we will just have a log message and continue, to eventually return no error. Because we don't need to retry for these cases.

Reply:
We probably want to iterate through all the webhooks before returning regardless of success of failure on any one webhook.

Otherwise persistent failure for one webhook would prevent all later webhooks from being updated.

Reply:
Ah right agreed, will make an update next week on this!

Reply:
@maxsmythe @ritazh PTAL now that it will only return after attempting to inject certs to all VWH and CRD conversion WHs.